### PR TITLE
ci: Run benchmarks with Ethernet and max MTUs

### PR DESCRIPTION
**Pull request by larseggert**

### CODE REVIEW

File: neqo-transport/benches/transfer.rs
Comment (mxinden) [inline neqo-transport/benches/transfer.rs:1]: Unrelated change? Feel free to leave it here, given that it is a benchmark only.

**Review by mxinden — APPROVED** [review]
apart from msquic failures, looks good to me.

File: neqo-transport/benches/transfer.rs
Comment (larseggert) [reply neqo-transport/benches/transfer.rs:1]: Ah yes, that fixes a new clippy warning.

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -132,50 +134,57 @@ jobs:
...
-                  taskset -c 1 nice -n -20 \
-                    perf $PERF_OPT -o "$client-$server$EXT.client.perf" \
-                      hyperfine -N --output null -w 1 -s "sleep 1" -n "$TAG" -u millisecond --export-markdown step.md "$CMD" |
-                      tee -a comparison.txt
-                  echo >> comparison.txt
-                  kill $PID
-                  cat step.md >> steps.md
-                  # Sanity check the size of the last retrieved file.
-                  [ "$(wc -c <"$SIZE")" -eq "$SIZE" ] || exit 1
+          # See https://github.com/microsoft/msquic/issues/4618#issuecomment-2422611592
```
Comment (martinthomson) [inline .github/workflows/bench.yml:137]: Why is this giant thing not a shell script?

File: neqo-transport/benches/transfer.rs [hunk 2]
```diff
@@ -63,14 +63,18 @@ fn benchmark_transfer(c: &mut Criterion, label: &str, seed: &Option<impl AsRef<s
...
+        std::env::var("SIMULATION_SEED").ok().as_ref(),
+    );
 }
 
 fn benchmark_transfer_fixed(c: &mut Criterion) {
     benchmark_transfer(
         c,
         "same-seed",
-        &Some("62df6933ba1f543cece01db8f27fb2025529b27f93df39e19f006e1db3b8c843"),
+        Some(&"62df6933ba1f543cece01db8f27fb2025529b27f93df39e19f006e1db3b8c843"),
```
Comment (martinthomson) [inline neqo-transport/benches/transfer.rs:77]: So you are passing `&&str` and changing the signature to `&impl AsRef`?  That seems a bit "wat" to me.

`impl AsRef<str>` matches `String` and `&str`.  Why does it need an extra `&`?  That only reduces ergonomics.

Comment (larseggert) [reply neqo-transport/benches/transfer.rs:77]: Clippy wants it

**Pull request merged**